bears/uploadTool.py

```diff
@@ -0,0 +1,79 @@
+import glob
+import os
+import re
+import shutil
+import sys
+
+# inserting path here so the following imports work wherever you run the script
+# in coala-bears
+sys.path.insert(0, '..')
+
+from coalib.collecting.Importers import iimport_objects
+from string import Template
+
+
+def create_directory(folder_name):
+    """
+    Checks if the folder exists, and creates it if it doesn't.
+
+    :param name: The directory name.
+    """
+    if not os.path.exists(folder_name):
+        os.mkdir(folder_name)
+
+
+def create_empty_file(file_name):
+    """
+    Creates an empty file.
+
+    :param file_name: Name of the file.
+    """
+    if not os.path.exists(file_name):
+        open(file_name, 'w').close()
+
+
+def main():
+    create_directory('upload')
+
+    bears_full_filenames = glob.glob('**/*Bear.py')
+
+    for bear_file_name in bears_full_filenames:
+        file_name = bear_file_name
+        bear_name = re.match(r'.+/(.*)\.py', bear_file_name).group(1)
+        bear_object = list(iimport_objects(file_name, attributes='kind',
+                                           local=True))[0]
+        create_directory(os.path.join('upload', bear_name))
+        create_directory(os.path.join('upload', bear_name, bear_name))
+        create_empty_file(os.path.join('upload', bear_name, bear_name,
+                                       '__init__.py'))
+        shutil.copyfile(file_name, os.path.join('upload', bear_name, bear_name,
+                                                bear_name + '.py'))
+
+        with open('setup.py.in') as fl:
+            template = Template(fl.read())
+
+        NAME = bear_name
+        VERSION = '1.0.30'  # testing purposes, not sure yet how we hold it
+        AUTHORS = str(bear_object.AUTHORS)
+        AUTHORS_EMAILS = str(bear_object.AUTHORS_EMAILS)
+        MAINTAINERS = str(bear_object.maintainers)
+        MAINTAINERS_EMAILS = str(bear_object.maintainers_emails)
+        PLATFORMS = str(bear_object.PLATFORMS)
+        LICENSE = str(bear_object.LICENSE)
+
+        processed_template = template.safe_substitute(**locals())
+
+        with open(os.path.join('upload', bear_name, 'setup.py'), "w") as setup:
+            setup.write(processed_template)
+
+            # upload everything, commented so I dont upload by mistake
```

Comment:
instead why not provide a flag? I have yet to find a legit reason for commented code. Still convinced there is none :)